src/algorithms/toolbox.jl

```diff
@@ -53,90 +53,140 @@ function cost_function(peps::InfinitePEPS, env::CTMRGEnv, O::LocalOperator)
 end
 
 function LinearAlgebra.norm(peps::InfinitePEPS, env::CTMRGEnv)
-    total = one(scalartype(peps))
-
-    for r in 1:size(peps, 1), c in 1:size(peps, 2)
-        rprev = _prev(r, size(peps, 1))
-        rnext = _next(r, size(peps, 1))
-        cprev = _prev(c, size(peps, 2))
-        cnext = _next(c, size(peps, 2))
-        total *= @autoopt @tensor env.edges[WEST, r, cprev][χ1 D1 D2; χ2] *
-            env.corners[NORTHWEST, rprev, cprev][χ2; χ3] *
-            env.edges[NORTH, rprev, c][χ3 D3 D4; χ4] *
-            env.corners[NORTHEAST, rprev, cnext][χ4; χ5] *
-            env.edges[EAST, r, cnext][χ5 D5 D6; χ6] *
-            env.corners[SOUTHEAST, rnext, cnext][χ6; χ7] *
-            env.edges[SOUTH, rnext, c][χ7 D7 D8; χ8] *
-            env.corners[SOUTHWEST, rnext, cprev][χ8; χ1] *
-            peps[r, c][d; D3 D5 D7 D1] *
-            conj(peps[r, c][d; D4 D6 D8 D2])
-        total *= tr(
-            env.corners[NORTHWEST, rprev, cprev] *
-            env.corners[NORTHEAST, rprev, c] *
-            env.corners[SOUTHEAST, r, c] *
-            env.corners[SOUTHWEST, r, cprev],
-        )
-        total /= @autoopt @tensor env.edges[WEST, r, cprev][χ1 D1 D2; χ2] *
-            env.corners[NORTHWEST, rprev, cprev][χ2; χ3] *
-            env.corners[NORTHEAST, rprev, c][χ3; χ4] *
-            env.edges[EAST, r, c][χ4 D1 D2; χ5] *
-            env.corners[SOUTHEAST, rnext, c][χ5; χ6] *
-            env.corners[SOUTHWEST, rnext, cprev][χ6; χ1]
-        total /= @autoopt @tensor env.corners[NORTHWEST, rprev, cprev][χ1; χ2] *
-            env.edges[NORTH, rprev, c][χ2 D1 D2; χ3] *
-            env.corners[NORTHEAST, rprev, cnext][χ3; χ4] *
-            env.corners[SOUTHEAST, r, cnext][χ4; χ5] *
-            env.edges[SOUTH, r, c][χ5 D1 D2; χ6] *
-            env.corners[SOUTHWEST, r, cprev][χ6; χ1]
-    end
+    return value(InfiniteSquareNetwork(peps), env)
```

Comment:
I think it makes sense to implemented the `norm` of an `InfinitePEPS` as the value of the corresponding norm network.